src/dotnet-tinify/ImageSquasher.cs

```diff
@@ -1,29 +1,30 @@
 ﻿using Humanizer;
 using McMaster.Extensions.CommandLineUtils;
+using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
 using TinifyAPI;
-using SysException = System.Exception;
 
 namespace ImageOptimiser
 {
-    public class ImageOptimiser
+    public class ImageSquasher
     {        
         public IConsole Console { get; }
 
-        public ImageOptimiser(IConsole console) => Console = console;
+        public ImageSquasher(IConsole console) => Console = console;
 
-        public async Task OptimiseFileAsync(string[] filesToSquash)
+        public async Task SquashFileAsync(IEnumerable<string> filesToSquash)
         {
-            if (filesToSquash?.Length > 0)
-            {
-                var compressFileTasks = filesToSquash
-                    .Where(file => Constants.SupportedExtensions.Contains(Path.GetExtension(file)))
-                    .Select(file => CompressFileAsync(file));
+            if (filesToSquash is null) throw new ArgumentNullException(nameof(filesToSquash));
+            if (!filesToSquash.Any()) throw new ArgumentException("There are no files to squash.", nameof(filesToSquash));
```

Comment:
Is this check necessary - I don't think we want to throw if we don't happen to find any files to squash? I assume `Task.WhenAll` will just complete if there's no files?

Reply:
Initially, I thought it might be a nice to know that there were no files -- instead of it quietly appearing to work but not actually taking any action. I removed it though.